src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -716,6 +716,7 @@ public synchronized StringBuffer reverse() {
     @Override
     public synchronized StringBuffer repeat(int codePoint, int count) {
         super.repeat(codePoint, count);
+        toStringCache = null;
```

Comment:
The other cases in StringBuffer clear toStringCache *before* performing the modification.
For consistency, I'd suggest doing the same.

Reply:
Changed